File: backend/schemas/route.py
from pydantic import BaseModel, Field
from typing import Optional
from .location import Location

class RouteBase(BaseModel):
    origin_location_id: int = Field(..., description="ID of the origin location")
    destination_location_id: int = Field(..., description="ID of the destination location")
    distance: float = Field(..., gt=0, description="Distance in kilometers", example=240.5)
    duration: float = Field(..., gt=0, description="Duration in hours", example=4.5)
    price: float = Field(..., gt=0, description="Price in bolivianos", example=35.0)

class RouteCreate(RouteBase):
    """
    Schema for creating a new route
    """
    pass

class RouteUpdate(BaseModel):
    """
    Schema for updating a route
    """
    origin_location_id: Optional[int] = None
    destination_location_id: Optional[int] = None
    distance: Optional[float] = Field(None, gt=0)
    duration: Optional[float] = Field(None, gt=0)
    price: Optional[float] = Field(None, gt=0)

class Route(RouteBase):
    """
    Schema for reading a route, includes relationships
    """
    id: int
    origin_location: Location
    destination_location: Location

    class Config:
        from_attributes = True

File: frontend/src/types/route.ts
import type { Location } from './location'

export interface Route {
  id: number
  origin_id?: number
  destination_id?: number
  origin_location_id?: number
  destination_location_id?: number
  base_price?: number
  price?: number
  estimated_duration?: number
  distance?: number
  duration?: number
  is_active?: boolean
  origin?: Location
  destination?: Location
  origin_location?: Location
  destination_location?: Location
  schedules?: RouteSchedule[]
  [key: string]: unknown
}

export interface RouteSchedule {
  id: number
  route_id: number
  departure_time: string
  arrival_time?: string
  days_of_week?: string[]
  is_active?: boolean
  [key: string]: unknown
}

export interface RouteWithSchedules extends Route {
  schedules: RouteSchedule[]
}
